articles/introduction-to-android-views-and-view-groups/index.md

```diff
@@ -0,0 +1,352 @@
+- `app:layout_constraintBottom_toBottomof="parent"` - This attribute constrains the bottom of the EditText to the end of another view which in this case is the parent.
+- `app:layout_constraintEnd_toEndof="parent"` - This attribute constrains the end of the EditText to the end of another view which in this case is the parent.
+- `app:layout_constraintTop_toTopof="parent"` - This attribute constrains the top of the EditText to the top of another view which in this case is the parent.
+- `android:layout_marginStart="20dp"` - sets the distance from start between the constrains to 20dp.
+- `android:layout_marginLeft="20dp"` - sets the distance from left between the constrains to 20dp
+- `android:layout_marginTop="20dp"` - sets distance from top between the constraints to 20dp
+- `app:layout_constraintStart_toStartof="parent"` - This attribute constrains the start of the button to the start of another view which in this case is the parent.
+- `app:layout_constraintTop_toTopof="+@id/editText"` - This attribute constrains the top of the button to the top of another view which in this case is the EditText.
+- `android:layout_marginBottom="20dp"` - sets the bottom distance between the constrains to 20dp.
+- `app:layout_constraintBottom_toBottomof="parent"` - This attribute constrains the bottom of the ImageView to the bottom of another view which in this case is the parent.
+- `app:layout_constraintEnd_toEndof="parent"` - This attribute constrains the end of the ImageView to the end of another view which in this case is the parent.
+- `app:layout_constraintStart_toStartof="parent"` - This attribute constrains the start of the ImageView to the start of another view which in this case is the parent.
+
+
+**FrameLayout**
+FrameLayout holds a single child view. it is designed to block out a region on the screen to view a single item.
+
+```xml
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".MainActivity">
+
+    <TextView
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_gravity="center"
+        android:layout_marginTop="20dp"
+        android:text="hello world"
+        android:textColor="#df3e4c"
+        android:textSize="20sp" />
+
+
+</FrameLayout>
+```
+![Output screen](/engineering-education/android-views-and-viewgroups/screenshot-frame-layout.png)
+
+The attributes above can be defined as follows:
+- `android:layout_marginTop="20dp` - Specifies extra space on the top side of this view. Hence, the extra space is set to "20dp"
+- `android:layout_gravity="center"` - controls how frame layout aligns all the views it contains. The value you set affects both horizontal and vertical alignment of all child views within the single row or column. In this case, the views are aligned at the center.
```

Comment:
controls how frame layout aligns the view it contains. not all the views. It is only one.